### PR TITLE
Set up project and analytics logic

### PR DESCRIPTION
**Pull request by etanshaul**
implementation as detailed in go/ide-analytics-as-a-library

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CompilerConfiguration">
+    <bytecodeTargetLevel>
+      <module name="ide-metrics-common_main" target="1.8" />
+      <module name="ide-metrics-common_test" target="1.8" />
+    </bytecodeTargetLevel>
+  </component>
+</project>
```
Comment (ivanporty) [inline .idea/compiler.xml]: should we fix those end of lines so Github won't complain?

File: src/main/java/com/google/cloud/tools/ide/analytics/UsageTracker.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      return new GoogleUsageTracker(settings);
+    }
+
+    return new NoOpUsageTracker();
+  }
+
+  /**
+   * Returns a fluent API for pinging tracking events.
+   *
+   * <p>Example: {@code sendEvent("appengine.deployment").withLabel("flex").setValue(1).ping();}
```
Comment (ivanporty) [inline src/main/java/com/google/cloud/tools/ide/analytics/UsageTracker.java]: `trackEvent()` in the javadoc?

File: .idea/compiler.xml (hunk 1 above)
Comment (etanshaul) [reply .idea/compiler.xml]: done

File: src/main/java/com/google/cloud/tools/ide/analytics/UsageTracker.java (hunk 2 above)
Comment (etanshaul) [reply src/main/java/com/google/cloud/tools/ide/analytics/UsageTracker.java]: fixed. good catch

**Review by ivanporty — APPROVED** [review]

**Pull request merged**